src/Columns/ColumnVector.cpp

```diff
@@ -90,7 +90,7 @@ void ColumnVector<T>::updateWeakHash32(WeakHash32 & hash) const
 
     while (begin < end)
     {
-        *hash_data = intHashCRC32(*begin, *hash_data);
```

Comment:
Here was a mistake - floating point numbers were rounded to integers before hashing. Which still keeps the algorithm correct but makes it less optimal.